### ISSUE TITLE
Error: Wrong parameters for ServerErrorResponseException

### ISSUE THREAD

**Issue by tiemevanveen** [elastic/elasticsearch-php#476]
I'm getting a the following error when feeding NULL to elastic as the mapping for my Index..

```sh
PHP Fatal error:  Uncaught Error: Wrong parameters for Elasticsearch\Common\Exceptions\ServerErrorResponseException([string $message [, long $code [, Throwable $previous = NULL]]]) in /home/vagrant/vagrant/vendor/elasticsearch/elasticsearch/src/Elasticsearch/Connections/Connection.php:663
Stack trace:
#0 /home/vagrant/vagrant/vendor/elasticsearch/elasticsearch/src/Elasticsearch/Connections/Connection.php(663): Exception->__construct(Array, 500, Object(Elasticsearch\Common\Exceptions\ServerErrorResponseException))
#1 /home/vagrant/vagrant/vendor/elasticsearch/elasticsearch/src/Elasticsearch/Connections/Connection.php(636): Elasticsearch\Connections\Connection->tryDeserializeError(Array, 'Elasticsearch\\C...')
#2 /home/vagrant/vagrant/vendor/elasticsearch/elasticsearch/src/Elasticsearch/Connections/Connection.php(599): Elasticsearch\Connections\Connection->tryDeserialize500Error(Array)
#3 /home/vagrant/vagrant/vendor/elasticsearch/elasticsearch/src/Elasticsearch/Connections/Connection.php(265): Elasticsearch\Connections\Connecti in /home/vagrant/vagrant/vendor/elasticsearch/elasticsearch/src/Elasticsearch/Connections/Connection.php on line 663

Fatal error: Uncaught Error: Wrong parameters for Elasticsearch\Common\Exceptions\ServerErrorResponseException([string $message [, long $code [, Throwable $previous = NULL]]]) in /home/vagrant/vagrant/vendor/elasticsearch/elasticsearch/src/Elasticsearch/Connections/Connection.php on line 663

Error: Wrong parameters for Elasticsearch\Common\Exceptions\ServerErrorResponseException([string $message [, long $code [, Throwable $previous = NULL]]]) in /home/vagrant/vagrant/vendor/elasticsearch/elasticsearch/src/Elasticsearch/Connections/Connection.php on line 663

Call Stack:
    0.0006     352648   1. {main}() /home/vagrant/vagrant/src/Core/Cli/update_search.php:0
    0.3232    1830808   2. Elasticsearch\Namespaces\IndicesNamespace->create() /home/vagrant/vagrant/src/Core/Cli/update_search.php:13
    0.4169    2086224   3. Elasticsearch\Endpoints\AbstractEndpoint->resultOrFuture() /home/vagrant/vagrant/vendor/elasticsearch/elasticsearch/src/Elasticsearch/Namespaces/IndicesNamespace.php:950
    0.4169    2086224   4. GuzzleHttp\Ring\Future\FutureArray->wait() /home/vagrant/vagrant/vendor/elasticsearch/elasticsearch/src/Elasticsearch/Endpoints/AbstractEndpoint.php:175
```

It seems that your error handling code fails. I think the wrong parameter error is caused by PHP7. a `echo json_encode($error)` on [line 663 of `Connection.php`](https://github.com/elastic/elasticsearch-php/blob/2.0/src/Elasticsearch/Connections/Connection.php#L662) results in: 

```
{
    "error": {
        "root_cause": [
            {
                "type": "null_pointer_exception",
                "reason": null
            }
        ],
        "type": "null_pointer_exception",
        "reason": null
    },
    "status": 500
}
```

That's probably what you want to return right? 

This is of course my own fault, I shouldn't pass NULL as the mapping. But in my case the mapping was loaded from a json file with invalid json and It wasn't clear to me that that was causing the failure. Here's an example:

```php
$elastic = ClientBuilder::fromConfig(['hosts' => ["127.0.0.1:9200"]]);

$mapping = json_decode(...); // this somehow fails, resulting in NULL

$elastic->indices()->create([
    'index' => 'my-index',
    'body' => [
        'mappings' => $mapping
    ]
]);
```

Yeah, I know, a check of `$mapping` before passing  it to elastic would have been nice ;). But I mean there might be other cases where null is passed accidentally.. 

-------

OS: Ubuntu14.04
PHP 7.0
ES-PHP: 2.2.1
ES: 2.3.4

**Comment by polyfractal** [comment]
Ah, yes, I see.  It's because the NPE doesn't set a reason (NPEs never do), so the auto-detect heuristic used to handle both pre/post 2.0 structured errors fails to detect correctly.  Which makes it try and parse as unstructured, then barfs because the format is different.

Committing a fix right now, and also removing this heuristic in master since it isn't needed anymore.  Thanks for the report!

**Comment by polyfractal** [comment]
Merged, and tagged in `v2.2.2`.  Thanks!
